docs/source/cloud-setup/cloud-permissions/kubernetes.rst

```diff
@@ -141,94 +195,134 @@ To create a service account that has the necessary permissions for SkyPilot, you
-      verbs: ["get", "list", "watch"]
-    - apiGroups: ["networking.k8s.io"]   # Required for exposing services.
-      resources: ["ingressclasses"]
-      verbs: ["get", "list", "watch"]
+      - apiGroups: ["*"]
+        resources: ["*"]
+        verbs: ["*"]
     ---
-    # ClusterRoleBinding for the service account
+    # Optional: If using object store mounting, create rolebinding in the skypilot-system
+    # namespace to create FUSE device manager.
     apiVersion: rbac.authorization.k8s.io/v1
-    kind: ClusterRoleBinding
+    kind: RoleBinding
     metadata:
-      name: sky-sa-cluster-role-binding
-      namespace: default
+      name: sky-sa-skypilot-system-role-binding
+      namespace: skypilot-system  # Do not change this namespace
       labels:
-          parent: skypilot
+        parent: skypilot
+    subjects:
+      - kind: ServiceAccount
+        name: sky-sa  # Change to your service account name
+        namespace: default  # Change this to the namespace where the service account is created
+    roleRef:
+      kind: Role
+      name: skypilot-system-service-account-role
+      apiGroup: rbac.authorization.k8s.io
+    ---
+    # Optional: Role for accessing ingress resources
+    apiVersion: rbac.authorization.k8s.io/v1
+    kind: Role
+    metadata:
+      name: sky-sa-role-ingress-nginx
+      namespace: ingress-nginx  # Do not change this namespace
+    rules:
+      - apiGroups: [""]
+        resources: ["services"]
+        verbs: ["list", "get", "watch"]
+      - apiGroups: ["rbac.authorization.k8s.io"]
+        resources: ["roles", "rolebindings"]
+        verbs: ["list", "get", "watch"]
+    ---
+    # Optional: RoleBinding for accessing ingress resources
+    apiVersion: rbac.authorization.k8s.io/v1
+    kind: RoleBinding
+    metadata:
+      name: sky-sa-rolebinding-ingress-nginx
+      namespace: ingress-nginx  # Do not change this namespace
     subjects:
-    - kind: ServiceAccount
-      name: sky-sa
-      namespace: default
+      - kind: ServiceAccount
+        name: sky-sa  # Change to your service account name
+        namespace: default  # Change this to the namespace where the service account is created
     roleRef:
-        kind: ClusterRole
-        name: sky-sa-cluster-role
-        apiGroup: rbac.authorization.k8s.io
+      kind: Role
+      name: sky-sa-role-ingress-nginx
+      apiGroup: rbac.authorization.k8s.io
 
 Create the service account using the following command:
 
 .. code-block:: bash
 
     $ kubectl apply -f create-sky-sa.yaml
 
-After creating the service account, configure SkyPilot to use it through ``~/.sky/config.yaml``:
+After creating the service account, the cluster admin may distribute kubeconfigs with the ``sky-sa`` service account to users who need to access the cluster.
+
+Users should also configure SkyPilot to use the ``sky-sa`` service account through ``~/.sky/config.yaml``:
```

Comment:
Using the default service account `skypilot-service-account` requires additional permissions because our code is self-correcting - it automatically fixes any service account misconfigurations by creating/patching resources (e.g., when user is upgrading versions or has accidentally deleted k8s RBAC resources created by SkyPilot). 

1. Our code will inspect and create/update the [necessary roles and rolebindings](https://github.com/skypilot-org/skypilot/blob/d61f10a190b338664b9a261c7a27d369e04475be/sky/provision/kubernetes/config.py#L38-L48). This requires additional `"create", "patch"` permissions on `"clusterroles", "clusterrolebindings"`.
2. Our code will inspect and create the `skypilot-system` [namespace](https://github.com/skypilot-org/skypilot/blob/d61f10a190b338664b9a261c7a27d369e04475be/sky/provision/kubernetes/config.py#L63) if it doesn't exist. This requires additional `"list", "create"` permissions on `namespaces`. 

These extra permissions on `"clusterroles", "clusterrolebindings", "namespaces"` may be considered too permissive in some environments (e.g., shared clusters). Perhaps we should keep the permissions here limited to minimal permissions required?